// sarvamai/dist/cjs/api/types/ConfigureConnection.js
"use strict";
/**
 * This file was auto-generated by Fern from our API Definition.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ConfigureConnection = void 0;
var ConfigureConnection;
(function (ConfigureConnection) {
    let Data;
    (function (Data) {
        Data.TargetLanguageCode = {
            BnIn: "bn-IN",
            EnIn: "en-IN",
            GuIn: "gu-IN",
            HiIn: "hi-IN",
            KnIn: "kn-IN",
            MlIn: "ml-IN",
            MrIn: "mr-IN",
            OdIn: "od-IN",
            PaIn: "pa-IN",
            TaIn: "ta-IN",
            TeIn: "te-IN",
        };
        Data.Speaker = {
            Anushka: "anushka",
            Abhilash: "abhilash",
            Manisha: "manisha",
            Vidya: "vidya",
            Arya: "arya",
            Karun: "karun",
            Hitesh: "hitesh",
        };
        Data.OutputAudioCodec = {
            Mp3: "mp3",
            Linear16: "linear16",
            Mulaw: "mulaw",
            Alaw: "alaw",
            Opus: "opus",
            Flac: "flac",
            Aac: "aac",
            Wav: "wav",
        };
        Data.OutputAudioBitrate = {
            ThirtyTwoK: "32k",
            SixtyFourK: "64k",
            NinetySixK: "96k",
            OneHundredTwentyEightK: "128k",
            OneHundredNinetyTwoK: "192k",
        };
    })(Data = ConfigureConnection.Data || (ConfigureConnection.Data = {}));
})(ConfigureConnection || (exports.ConfigureConnection = ConfigureConnection = {}));
