// sarvamai/dist/cjs/Client.js
"use strict";
/**
 * This file was auto-generated by Fern from our API Definition.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.SarvamAIClient = void 0;
const core = __importStar(require("./core/index.js"));
const headers_js_1 = require("./core/headers.js");
const Client_js_1 = require("./api/resources/text/client/Client.js");
const Client_js_2 = require("./api/resources/speechToText/client/Client.js");
const Client_js_3 = require("./api/resources/textToSpeech/client/Client.js");
const Client_js_4 = require("./api/resources/chat/client/Client.js");
const Client_js_5 = require("./api/resources/speechToTextJob/client/Client.js");
const Client_js_6 = require("./api/resources/speechToTextTranslateJob/client/Client.js");
const Client_js_7 = require("./api/resources/speechToTextStreaming/client/Client.js");
const Client_js_8 = require("./api/resources/speechToTextTranslateStreaming/client/Client.js");
const Client_js_9 = require("./api/resources/textToSpeechStreaming/client/Client.js");
class SarvamAIClient {
    constructor(_options = {}) {
        this._options = Object.assign(Object.assign({}, _options), { headers: (0, headers_js_1.mergeHeaders)({
                "X-Fern-Language": "JavaScript",
                "X-Fern-SDK-Name": "sarvamai",
                "X-Fern-SDK-Version": "0.1.9",
                "User-Agent": "sarvamai/0.1.9",
                "X-Fern-Runtime": core.RUNTIME.type,
                "X-Fern-Runtime-Version": core.RUNTIME.version,
            }, _options === null || _options === void 0 ? void 0 : _options.headers) });
    }
    get text() {
        var _a;
        return ((_a = this._text) !== null && _a !== void 0 ? _a : (this._text = new Client_js_1.Text(this._options)));
    }
    get speechToText() {
        var _a;
        return ((_a = this._speechToText) !== null && _a !== void 0 ? _a : (this._speechToText = new Client_js_2.SpeechToText(this._options)));
    }
    get textToSpeech() {
        var _a;
        return ((_a = this._textToSpeech) !== null && _a !== void 0 ? _a : (this._textToSpeech = new Client_js_3.TextToSpeech(this._options)));
    }
    get chat() {
        var _a;
        return ((_a = this._chat) !== null && _a !== void 0 ? _a : (this._chat = new Client_js_4.Chat(this._options)));
    }
    get speechToTextJob() {
        var _a;
        return ((_a = this._speechToTextJob) !== null && _a !== void 0 ? _a : (this._speechToTextJob = new Client_js_5.SpeechToTextJob(this._options)));
    }
    get speechToTextTranslateJob() {
        var _a;
        return ((_a = this._speechToTextTranslateJob) !== null && _a !== void 0 ? _a : (this._speechToTextTranslateJob = new Client_js_6.SpeechToTextTranslateJob(this._options)));
    }
    get speechToTextStreaming() {
        var _a;
        return ((_a = this._speechToTextStreaming) !== null && _a !== void 0 ? _a : (this._speechToTextStreaming = new Client_js_7.SpeechToTextStreaming(this._options)));
    }
    get speechToTextTranslateStreaming() {
        var _a;
        return ((_a = this._speechToTextTranslateStreaming) !== null && _a !== void 0 ? _a : (this._speechToTextTranslateStreaming = new Client_js_8.SpeechToTextTranslateStreaming(this._options)));
    }
    get textToSpeechStreaming() {
        var _a;
        return ((_a = this._textToSpeechStreaming) !== null && _a !== void 0 ? _a : (this._textToSpeechStreaming = new Client_js_9.TextToSpeechStreaming(this._options)));
    }
}
exports.SarvamAIClient = SarvamAIClient;
