// sarvamai/dist/esm/Client.mjs
/**
 * This file was auto-generated by Fern from our API Definition.
 */
import * as core from "./core/index.mjs";
import { mergeHeaders } from "./core/headers.mjs";
import { Text } from "./api/resources/text/client/Client.mjs";
import { SpeechToText } from "./api/resources/speechToText/client/Client.mjs";
import { TextToSpeech } from "./api/resources/textToSpeech/client/Client.mjs";
import { Chat } from "./api/resources/chat/client/Client.mjs";
import { SpeechToTextJob } from "./api/resources/speechToTextJob/client/Client.mjs";
import { SpeechToTextTranslateJob } from "./api/resources/speechToTextTranslateJob/client/Client.mjs";
import { SpeechToTextStreaming } from "./api/resources/speechToTextStreaming/client/Client.mjs";
import { SpeechToTextTranslateStreaming } from "./api/resources/speechToTextTranslateStreaming/client/Client.mjs";
import { TextToSpeechStreaming } from "./api/resources/textToSpeechStreaming/client/Client.mjs";
export class SarvamAIClient {
    constructor(_options = {}) {
        this._options = Object.assign(Object.assign({}, _options), { headers: mergeHeaders({
                "X-Fern-Language": "JavaScript",
                "X-Fern-SDK-Name": "sarvamai",
                "X-Fern-SDK-Version": "0.1.9",
                "User-Agent": "sarvamai/0.1.9",
                "X-Fern-Runtime": core.RUNTIME.type,
                "X-Fern-Runtime-Version": core.RUNTIME.version,
            }, _options === null || _options === void 0 ? void 0 : _options.headers) });
    }
    get text() {
        var _a;
        return ((_a = this._text) !== null && _a !== void 0 ? _a : (this._text = new Text(this._options)));
    }
    get speechToText() {
        var _a;
        return ((_a = this._speechToText) !== null && _a !== void 0 ? _a : (this._speechToText = new SpeechToText(this._options)));
    }
    get textToSpeech() {
        var _a;
        return ((_a = this._textToSpeech) !== null && _a !== void 0 ? _a : (this._textToSpeech = new TextToSpeech(this._options)));
    }
    get chat() {
        var _a;
        return ((_a = this._chat) !== null && _a !== void 0 ? _a : (this._chat = new Chat(this._options)));
    }
    get speechToTextJob() {
        var _a;
        return ((_a = this._speechToTextJob) !== null && _a !== void 0 ? _a : (this._speechToTextJob = new SpeechToTextJob(this._options)));
    }
    get speechToTextTranslateJob() {
        var _a;
        return ((_a = this._speechToTextTranslateJob) !== null && _a !== void 0 ? _a : (this._speechToTextTranslateJob = new SpeechToTextTranslateJob(this._options)));
    }
    get speechToTextStreaming() {
        var _a;
        return ((_a = this._speechToTextStreaming) !== null && _a !== void 0 ? _a : (this._speechToTextStreaming = new SpeechToTextStreaming(this._options)));
    }
    get speechToTextTranslateStreaming() {
        var _a;
        return ((_a = this._speechToTextTranslateStreaming) !== null && _a !== void 0 ? _a : (this._speechToTextTranslateStreaming = new SpeechToTextTranslateStreaming(this._options)));
    }
    get textToSpeechStreaming() {
        var _a;
        return ((_a = this._textToSpeechStreaming) !== null && _a !== void 0 ? _a : (this._textToSpeechStreaming = new TextToSpeechStreaming(this._options)));
    }
}
